### PR TITLE
use the actual retention duration

### PR DESCRIPTION
**Pull request by pinux-studio**
when  retainingDataOnFailure is set to true

### CODE REVIEW

**Comment by pinux-studio** [comment]
@pettermahlen

**Comment by Korev** [comment]
:+1:

File: src/test/java/com/spotify/dns/DnsSrvResolversIT.java [hunk 1]
```diff
@@ -74,6 +75,16 @@ public void shouldFailForBadHostNames() throws Exception {
     }
   }
 
+  @Test
+  public void shouldSucceedCreatingRetainingDnsResolver() throws Exception {
+    try {
+      resolver = DnsSrvResolvers.newBuilder().retainingDataOnFailures(true).build();
+    }
+    catch (DnsException e) {
```
Comment (pettermahlen) [inline src/test/java/com/spotify/dns/DnsSrvResolversIT.java]: No need to even catch this exception - just validate that construction is possible without throwing anything. That would have been enough to catch this particular mistake. So it'll be enough with something like:

```
public void shouldSucceedCreatingRetainingDnsResolver() throws Exception {
  // validate that construction of a retaining resolver is possible
  DnsSrvResolvers.newBuilder().retainingDataOnFailures(true).build();
}
```

If I'm right, that test would have failed before your fix?

Comment (pinux-studio) [inline src/test/java/com/spotify/dns/DnsSrvResolversIT.java]: yes

**Pull request merged**